frontend/src/pages/projects/projectSharing/ProjectSharingTableRow.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import { ActionsColumn, Tbody, Td, Tr } from '@patternfly/react-table';
+import {
+  Button,
+  DropdownDirection,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Text,
+} from '@patternfly/react-core';
+import { CheckIcon, TimesIcon } from '@patternfly/react-icons';
+import { RoleBindingKind } from '~/k8sTypes';
+import { castProjectSharingRoleType, formatDate } from './utils';
+import { ProjectSharingRoleType } from './types';
+
+type ProjectSharingTableRowProps = {
```

Comment:
I'm not happy with this component, I tried to follow patternfly solution for table editing but it's not working, there are only working solutions with the old `<Table>` component, there might be a better solution for this.